include/libcyphal/transport/udp/delegate.hpp

```diff
@@ -69,7 +75,150 @@ struct AnyUdpardTxMetadata
 ///
 class TransportDelegate
 {
+    // FD977E25-CD36-48C5-B02E-B31420DCFF3B
+    using UdpardMemoryTypeIdType = cetl::
+        // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+        type_id_type<0xFD, 0x97, 0x7E, 0x25, 0xCD, 0x36, 0x48, 0xC5, 0xB0, 0x2E, 0xB3, 0x14, 0x20, 0xDC, 0xFF, 0x3B>;
+
 public:
+    /// @brief RAII class to manage memory allocated by Udpard library.
+    ///
+    /// NOSONAR cpp:S4963 for below `class UdpardMemory` - we do directly handle resources here.
+    ///
+    class UdpardMemory final  // NOSONAR cpp:S4963
+        : public cetl::rtti_helper<UdpardMemoryTypeIdType, ScatteredBuffer::IStorage>
+    {
+    public:
+        UdpardMemory(TransportDelegate& delegate, UdpardRxTransfer& transfer)
+            : delegate_{delegate}
+            , payload_size_{std::exchange(transfer.payload_size, 0)}
+            , payload_{std::exchange(transfer.payload, {})}
+        {
+        }
+        UdpardMemory(UdpardMemory&& other) noexcept
+            : delegate_{other.delegate_}
+            , payload_size_{std::exchange(other.payload_size_, 0)}
+            , payload_{std::exchange(other.payload_, {})}
+        {
+        }
+        UdpardMemory(const UdpardMemory&) = delete;
+
+        ~UdpardMemory() override
+        {
+            ::udpardRxFragmentFree(payload_, delegate_.memoryResources().fragment, delegate_.memoryResources().payload);
+        }
+
+        UdpardMemory& operator=(const UdpardMemory&)     = delete;
+        UdpardMemory& operator=(UdpardMemory&&) noexcept = delete;
+
+        // MARK: ScatteredBuffer::IStorage
+
+        CETL_NODISCARD std::size_t size() const noexcept override
+        {
+            return payload_size_;
+        }
+
+        CETL_NODISCARD std::size_t copy(const std::size_t offset_bytes,
+                                        cetl::byte* const destination,
+                                        const std::size_t length_bytes) const override
+        {
+            using FragSpan = const cetl::span<const cetl::byte>;
+
+            // TODO: Use `udpardGather` function when it will be available with offset support.
```

Comment:
```suggestion
            // TODO: Use `udpardGather` function when it will be available with offset support.
            // See https://github.com/OpenCyphal/libudpard/issues/57
```